wallet-new/test/unit/UTxO/Interpreter.hs

```diff
@@ -499,35 +597,60 @@ instance DSL.Hash h Addr => Interpret h (DSL.Block h Addr) where
         -- figure out who needs to sign the block
         BlockSignInfo{..} <- asks $ blockSignInfoForSlot leaders slotId
 
-        withProtocolMagic $ \pm ->
-          withConfig $
-            createMainBlockPure
-              pm
-              blockSizeLimit
-              prev
-              (Just (bsiPSK, bsiLeader))
-              slotId
-              bsiKey
-              (RawPayload
-                  (toList ts)
-                  (defaultSscPayload (siSlot slotId)) -- TODO
-                  dlgPayload
-                  updPayload
-                )
+        pm <- asks magic
+        createMainBlockPure
+          pm
+          blockSizeLimit
+          prev
+          (Just (bsiPSK, bsiLeader))
+          slotId
+          bsiKey
+          (RawPayload
+              (toList ts)
+              (defaultSscPayload (siSlot slotId)) -- TODO
+              dlgPayload
+              updPayload
+            )
 
       -- TODO: Get this value from somewhere rather than hardcoding it
       blockSizeLimit = 2 * 1024 * 1024 -- 2 MB
 
 instance DSL.Hash h Addr => Interpret h (DSL.Chain h Addr) where
   type Interpreted (DSL.Chain h Addr) = OldestFirst [] Block
 
-  int :: forall e m. (Monad m)
+  int :: forall e m. Monad m
       => DSL.Chain h Addr -> IntT h e m (OldestFirst [] Block)
-  int (OldestFirst blocks) = OldestFirst . join <$>
-      mapM (fmap blockWithBoundary . int) blocks
+  int (OldestFirst blocks) = OldestFirst <$>
+      stripFinalEbb . concatMap flatten <$> mapM int blocks
     where
-      blockWithBoundary (UnsafeRawResolvedBlock block Nothing _) = [Right block]
-      blockWithBoundary (UnsafeRawResolvedBlock block (Just ebb) _ ) = [Left ebb, Right block]
+      flatten :: (RawResolvedBlock, Maybe GenesisBlock) -> [Block]
+      flatten (b, Nothing)  = [Right (rawResolvedBlock b)]
+      flatten (b, Just ebb) = [Right (rawResolvedBlock b), Left ebb]
+
+      -- The Cardano verifier has a curious pecularity that we are not allowed to
```

Comment:
Should clarify whether this is a property of the Cardano verifier, or whether it's just the `ESREmptyEpoch` case. And then discuss what we _should_ do in that case.

Reply:
I'll leave that in your capable hands :)